### PR TITLE
fix: update syndesis-loading to accept a TemplateRef too

### LINKED ISSUE #3587: [UI] syndesis-loading children is evaluated regardless of its loading state
## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Documentation issue or request
</code></pre>

## Description

Children elements of the `syndesis-loading` component are parsed regardless of the loading state of the component.

A simple example to try it out is the following:

```
// example.component.ts
export class Example {
   alert = alert;
}

// examle.component.html
<syndesis-loading [loading]="true">
  {{ alert("sorry, still evaluated!") }}
</syndesis-loading>
```

## Workaround

The following code does what you'd expect it to do.

```
// example.component.ts
export class Example {
   alert = alert;
}

// examle.component.html
<ng-container *ngIf="true; else content">
  <syndesis-loading loading="true"></syndesis-loading>
</ng-container>

<ng-template #content>
  {{ alert("this will not be called!") }}
</ng-template>
```

### PR DESCRIPTION
**Pull request by gashcrumb**
and update all the code using it.

fixes #3587 now code can pass a template ref to `syndesis-loading` using the `content` input:

```
<syndesis-loading [loading]="whatevs" [content]="ref">
  <ng-template #ref>
    <span>Hello! {{ stuff("world!") }}</span>
  </ng-template>
</syndesis-loading>
```

In this case `stuff()` doesn't get called until `whatevs` is false.

### CODE REVIEW

File: app/ui/src/app/common/loading/loading.component.html [hunk 1]
```diff
@@ -1,6 +1,11 @@
-<ng-container *ngIf="loading">
+<ng-container *ngIf="loading; else body">
```
Comment (riccardo-forina) [inline app/ui/src/app/common/loading/loading.component.html]: I think you can get away with just 

```
<ng-container *ngIf="loading; else content">
  <div class="syn-loading">
      <div class="spinner spinner-lg loading"></div>
  </div>
</ng-container>
```

File: app/ui/src/app/common/loading/loading.component.ts [hunk 2]
```diff
@@ -7,4 +7,5 @@ import { Component, Input } from '@angular/core';
 })
 export class LoadingComponent {
   @Input() loading: boolean;
+  @Input() content;
```
Comment (riccardo-forina) [inline app/ui/src/app/common/loading/loading.component.ts]: `@Input() content: TemplateRef<any>;`

File: app/ui/src/app/common/loading/loading.component.html (hunk 1 above)
Comment (gashcrumb) [reply app/ui/src/app/common/loading/loading.component.html]: I think I could but I kinda wanted to keep support for *not* using an *ng-template* still also.

File: app/ui/src/app/common/loading/loading.component.ts (hunk 2 above)
Comment (gashcrumb) [reply app/ui/src/app/common/loading/loading.component.ts]: Yep, good call

File: app/ui/src/app/common/loading/loading.component.html (hunk 1 above)
Comment (riccardo-forina) [reply app/ui/src/app/common/loading/loading.component.html]: Oh, I didn't think about that, you're right!

**Comment by gashcrumb** [comment]
Updated, but we can hold off merging until *after* the sprint demo just to be on the safe side :-)

**Pull request merged**